test/core/templaters/dbt_test.py

```diff
@@ -65,6 +67,57 @@ def test__templater_dbt_templating_result(
     assert str(templated_file) + "\n" == open("../dbt/" + fname).read()
 
 
+@pytest.mark.dbt
+def test__templater_dbt_templating_test_lex(in_dbt_project_dir, dbt_templater):
+    lexer = Lexer(config=FluffConfig(configs=DBT_FLUFF_CONFIG))
+    templated_file, _ = dbt_templater.process(
+        in_str="",
+        fname="tests/test.sql",
+        config=FluffConfig(configs=DBT_FLUFF_CONFIG),
+    )
+    tokens, lex_vs = lexer.lex(templated_file)
+
+
+@pytest.mark.dbt
+@pytest.mark.parametrize(
+    "raw_file,templated_file,result",
+    [
+        (
+            "select * from a",
+            """
+with dbt__CTE__INTERNAL_test as (
+select * from a
+)select count(*) from dbt__CTE__INTERNAL_test
+""",
+            [
+                ("literal", slice(0, 15, None), slice(35, 50, None)),

```

Comment:
@alanmcruickshank Are these slices what you'd expect from the above raw_file and templated_file?

Reply:
...I don't think so. I'm not 100% sure, but I think I would expect something more like:

```python
("templated", slice(0, 0, None), slice(0, 35, None)),
("literal", slice(0, 15, None), slice(35, 50, None)),
("templated", slice(15, 15, None), slice(50, 96, None))
```

What I'm not sure is whether we should deal with them as `templated` sections, or as something different, because they're not really `templated` in the true sense, more of a `wrapper`.  The current slicing that you're describing isn't working well with query wrapping in any case.